MUICore/muicore/mui_server.py

```diff
@@ -93,11 +98,64 @@ def find_f(state: StateBase):
             for addr in self.find:
                 m.add_hook(addr, find_f)
 
-            def manticore_runner(mcore: Manticore):
+            def manticore_native_runner(mcore: Manticore):
                 mcore.run()
                 mcore.finalize()
 
-            mthread = Thread(target=manticore_runner, args=(m,), daemon=True)
+            mthread = Thread(target=manticore_native_runner, args=(m,), daemon=True)
+            mthread.start()
+            self.manticore_instances[id] = (m, mthread)
+
+        except Exception as e:
+            print(e)
+            raise e
+            return ManticoreInstance()
+        return ManticoreInstance(uuid=id)
+
+    def StartEVM(
+        self, evm_arguments: EVMArguments, context: _Context
+    ) -> ManticoreInstance:
+        """Starts a singular Manticore instance to analyze a solidity contract"""
+        id = uuid.uuid4().hex
+        try:
+            m = ManticoreEVM()
+
+            args = setup_detectors_flags(
+                evm_arguments.detectors_to_exclude, evm_arguments.additional_flags, m
+            )
+
+            print(Path.cwd())
+
+            def manticore_evm_runner(m: ManticoreEVM, args: argparse.Namespace):
+                m.multi_tx_analysis(
+                    evm_arguments.contract_path,
+                    contract_name=evm_arguments.contract_name,
+                    tx_limit=-1
+                    if not evm_arguments.tx_limit
+                    else evm_arguments.tx_limit,
+                    tx_use_coverage=True
+                    if args.txnocoverage == None
+                    else args.txnocoverage,
+                    tx_send_ether=True if args.txnoether == None else args.txnoether,
+                    tx_account="attacker"
+                    if not evm_arguments.tx_account
+                    else evm_arguments.tx_account,
+                    tx_preconstrain=False
+                    if args.txpreconstrain == None
+                    else args.txpreconstrain,
+                    compile_args={
+                        "solc_solcs_bin": str(Path.cwd() / "solc")
+                        if not evm_arguments.solc_bin
+                        else evm_arguments.solc_bin
+                    },
```

Comment:
There should be some checks here whether `$(pwd)/solc` exists, and if it doesn't try checking for [`shutil.which('solc')`](https://docs.python.org/3.7/library/shutil.html#shutil.which)

Reply:
done! it'll now use `evm_arguments.solc_bin` if supplied, else attempts to locate `solc` on PATH. 

It no longer automatically tries to locate the "packaged" solc binary - I realised that trying to use `pwd` isn't really suitable since a MUI Plugin will probably be calling the muicore binary from a different location, and using `inspect.getframeinfo` like i did in #20 won't work since it points to the temporary python files created by shiv rather than the shiv binary itself. 

Instead, I think it's ok to have the MUI Plugins themselves do the work of locating the "packaged" solc binary and having it be a sensible default in the run/setup dialogs